corehq/apps/users/management/commands/add_download_reports_permission.py

```diff
@@ -9,7 +9,9 @@ class Command(BaseCommand):
     def handle(self, **options):
         permission, created = SQLPermission.objects.get_or_create(value='download_reports')
         num_roles_modified = 0
-        for role in SQLUserRole.objects.exclude(rolepermission__permission_fk_id=permission.id).iterator():
+        all_roles = SQLUserRole.objects.all()
+        roles_with_new_permission = SQLUserRole.objects.filter(rolepermission__permission_fk_id=permission.id)
+        for role in all_roles.difference(roles_with_new_permission).iterator():
```

Comment:
```suggestion
        all_role_ids = set(SQLUserRole.objects.all().values_list("id", flat=True))
        role_ids_with_new_permission = set(SQLUserRole.objects.filter(rolepermission__permission_fk_id=permission.id).value_list("id", flat=True))
        difference = all_role_ids.difference(role_ids_with_new_permission)
        for chunk in chunked(difference):
```

Reply:
I made change here for chunks of 50: 00968ac

Reply:
@snopoke does this change look like what you had in mind?